test/e2e/utils/httpClient.go

```diff
@@ -85,16 +82,13 @@ func (c *httpClientConfig) Post(path, payload string, headers map[string]string)
 
 func (c *httpClientConfig) exec(req *http.Request, headers map[string]string) (*http.Response, error) {
 	if c.isAuthRequired() {
-		if c.authRealm == nil {
-			rsp, err := c.Do(req)
-			ExpectNoError(err)
-			if rsp.StatusCode != http.StatusUnauthorized {
-				return rsp, fmt.Errorf("Expected 401 DIGEST response before content: %v", rsp)
-			}
-			c.authRealm = getAuthorization(*c.username, *c.password, rsp)
+		rsp, err := c.Do(req)
```

Comment:
@ryanemerson probably we make `rsp, err := c.Do(req)` calls twice and didn't close first `rsp` and this is the reason why after CR deletion connection still open and tries to invoke on the finish

Reply:
Good point, I had missunderstood what you meant about the closing of requests previously. Updated

Reply:
I think I finally have it :sweat_smile: 

I have added a DEBUG constant to `httpClient.go` that dumps all http req/rsp headers.